internal/webhooks/cluster.go

```diff
@@ -443,3 +470,40 @@ func machineDeploymentClassOfName(clusterClass *clusterv1.ClusterClass, name str
 	}
 	return nil
 }
+
+// validateCIDRBlocks validates that:
+// 1) No more than two CIDR blocks are specified.
```

Comment:
Is this a Kubernetes limitation that applies both to service and pod cidr?

Reply:
If not we are not sure I will simply validate all the CIDR are valid, without making assumption on the combination of them

Reply:
Both

Services: https://github.com/kubernetes/kubernetes/blob/6dc81c3280996a9a4267bf8ef30d4bda36f8a293/cmd/kube-apiserver/app/options/validation.go#L47-L69

Pods:  https://github.com/kubernetes/kubernetes/blob/08749750a96fe7f236befa40076a3117e7b36733/staging/src/k8s.io/cloud-provider/app/core.go#L127-L133

(see https://github.com/kubernetes-sigs/cluster-api/pull/7420#issuecomment-1290764699)